flask/app/routes.py

```diff
@@ -340,3 +341,191 @@ def get_enums():
         if issubclass(obj, Enum) and name != 'Enum':
             enums[name] = [e.value for e in getattr(models, name)]
     return jsonify(enums)
+
+
+# TODO: text/csv content type does not seem to be working, ie. request.files['file'] throws a KeyError
+# TODO: decorator for the try...catch.. statements? 
+# TODO: flesh out the fields to be updated for each entity from Hillary's slack message
+# TODO: enum errors are not super informative rn, eg for condition, mentions the field was truncated in the row
```

Comment:
Can you make use of the mixin function above that should take care of this problem?

Reply:
It looks like we can’t re-use it exactly - `json_mixin` takes as input, an existing record (as an object). If we check for inconsistencies before the record is added, then it fails as there is no existing record from the query. If we check for the inconsistency after the record is added in, this does not work as `flush()` will catch the (less informative) error first. It also edits the record so there's that too. What do you think?

Reply:
Let's extract the part that checks enum values then?